File: Source/Aladdin.CAPI.STB.Java/src/aladdin/capi/stb/stb11762/BDHPrivateKey.java
package aladdin.capi.stb.stb11762;
import aladdin.capi.*; 
import java.math.*; 

///////////////////////////////////////////////////////////////////////////
// Личный ключ алгоритма СТБ 1176.2
///////////////////////////////////////////////////////////////////////////
public class BDHPrivateKey extends aladdin.capi.PrivateKey implements IBDHPrivateKey
{
    // номер версии для сериализации
    private static final long serialVersionUID = 5316251136938809592L;
    
    // параметры ключа и секретное значение
    private final IBDHParameters parameters; private final BigInteger x;
    
    // конструктор
	public BDHPrivateKey(Factory factory, SecurityObject scope, 
        String keyOID, IBDHParameters parameters, BigInteger x) 
	{ 	
        // сохранить переданные параметры
		super(factory, scope, keyOID); this.parameters = parameters; this.x = x; 
    } 
    // параметры ключа
	@Override public final IBDHParameters parameters() { return parameters; } 
    // секретное значение
	@Override public final BigInteger bdhX() { return x; } 
}

File: Source/Aladdin.CAPI.STB/CAPI/STB/STB11762/BDHPrivateKey.cs
using System;
using System.Diagnostics.CodeAnalysis;

///////////////////////////////////////////////////////////////////////////
// Личный ключ алгоритма СТБ 1176.2
///////////////////////////////////////////////////////////////////////////
namespace Aladdin.CAPI.STB.STB11762
{
    [SuppressMessage("Microsoft.Design", "CA1063:ImplementIDisposableCorrectly")]
    public class BDHPrivateKey : CAPI.PrivateKey, IBDHPrivateKey
    {
        // параметры ключа и секретное значение
        private IBDHParameters parameters; private Math.BigInteger x;
    
        // конструктор
	    public BDHPrivateKey(CAPI.Factory factory, SecurityObject scope, string keyOID, 
            IBDHParameters parameters, Math.BigInteger x) : base(factory, scope, keyOID)
	    { 	
            // сохранить переданные параметры
		    this.parameters = parameters; this.x = x; 
        } 
        // параметры ключа
	    public override CAPI.IParameters Parameters { get { return parameters; }}
        // секретное значение
	    public virtual Math.BigInteger X { get { return x; }}
    }
}
